chapter_6/2_dictionary_learning.tex

```diff
@@ -0,0 +1,428 @@
+\begin{equation}
+\begin{aligned}
+|a'_{ij}-a^*_{ij}|\leq&\left(1-\eta\frac s r \mathbb E[x_j^2|x_j^*\neq0]\right)|a_{ij}-a^*_{ij}|+\eta[|E_1|+|E_2|+|\epsilon_n|]\\
+\leq&\left(1-\eta\frac s r\mathbb E[x_j^2|x_j^*\neq 0]\right)|a_{ij}-a_{ij}^*|+\eta\left(\frac s {3r} R\right)+\eta|\epsilon_n|\\
+\leq&\left(1-\eta\frac s r\left[\mathbb E[x_j^2|x_j^*\neq0]-\frac 1 3\right]\right)R+\eta|\epsilon_n|.
+\end{aligned}
+\end{equation}
+
+Now we consider the bound of the term $[\mathbb E[x_j^2|x_j^*\neq0]$. By the Lemma 6 in \cite{chatterji2017alternating}, we have:
+\begin{equation}
+    0\leq\left(1-\eta\frac s r\left[\mathbb E[x_j^2|x_j^*\neq0]-\frac 1 3\right]\right)\leq\frac 3 4
+\end{equation}
+therefore,
+\begin{equation}
+|a_{ij}^{(t)}-a_{ij}^*|\leq\frac3 4 R^{(t-1)}+\eta|\epsilon_n|\leq\frac3 4 R^{(t-1)}+\frac1 8 R^{(t-1)}=\frac 7 8R^{(t-1)}\leq\left(\frac 7 8\right)^tR^{(0)}.
+\end{equation}
+
+We have $\eta|\epsilon_n|\leq R/8\leq R^{(0)}/8$ with probability at least $1-\delta$ in each iteration. Thus by taking a union bound over the iterations we are guaranteed to remain in our initial ball of radius $R^{(0)}$ with high probability which completes the proof.
+
+\subsection{Alternating Minimization Algorithm II}
+In this section we describe another alternating minimization algorithm used in the paper \cite{ref1} to learn the dictionary matrix $A^*$. This paper implements another specific kind of alternating minimization - estimating coefficients via $\ell_1$ minimization and estimating the dictionary via least square estimation. The paper shows a theoretical guarantee for the above alternative minimization algorithm. The paper shows that under a set of conditions of the dictionary matrix $A^*$, coefficient matrix $X^*$ and sample complexity, we can show that the algorithm 1) satisfies local linear convergence, and 2) produces exact recovery of the dictionary matrix $A^*$ up to arbitrarily close. In fact, the paper shows that under the set of specified conditions, we need at most $\mathcal{O}(\log_2 1/\epsilon)$ rounds of alternating minimization to ensure that each dictionary atom vector is approximated within an absolute error of $\pm \epsilon$. In later sections, the paper proved a nice theoretical guarantee for this algorithm. We use $A(i)$ and $X(i)$ to denote the dictionary estimate and coefficient estimate after the $i$-th round of alternating minimization.  \\
+
+Input: Samples $Y$, initial dictionary $A(0)$, accuracy sequence $\epsilon_t$ and sparsity level $s$. (The way to estimate $A(0)$ and the selection of $\epsilon_t$'s will be explained later in the paper.)
+
+\begin{algorithm}
+
+\begin{algorithmic}
+\caption{Alternating Minimization for Dictionary Learning}
+
+\FOR{iterations $t=0, 1, 2, ..., T-1$}
+
+\FOR{samples $i=1, 2, ..., n$}
+
+$X(t+1)_i = \arg \min_{x \in \mathbb{R}^r}||x||_1$ such that $||Y_i - A(t)x||_2 \leq \epsilon_t$\\
+
+\ENDFOR
+
+Threshold: $X(t+1) = X(t+1) * (\mathbbm{1}[X(t+1) > 9s\epsilon_t])$\\
```

Comment:
Again there is a compilation error here. You need to add \STATE before each line I think.

Reply:
Yes.